kubernetes/apps_kubeclient.go

```diff
@@ -557,42 +558,42 @@ func (kc *kubeClient) getDeploymentConfig(namespace string, appName string, spac
 	dcURL := fmt.Sprintf("/oapi/v1/namespaces/%s/deploymentconfigs/%s", namespace, appName)
 	result, err := kc.getResource(dcURL, true)
 	if err != nil {
-		return nil, err
+		return nil, errs.WithStack(err)
 	} else if result == nil {
 		return nil, nil
 	}
 
 	// Parse deployment config from result
 	kind, ok := result["kind"].(string)
 	if !ok || kind != "DeploymentConfig" {
-		return nil, errors.New("No deployment config returned from endpoint")
+		return nil, errs.New("no deployment config returned from endpoint")
 	}
 	metadata, ok := result["metadata"].(map[interface{}]interface{})
 	if !ok {
-		return nil, errors.New("Metadata missing from deployment config")
+		return nil, errs.New("metadata missing from deployment config")
 	}
 	// Check the space label is what we expect
 	labels, ok := metadata["labels"].(map[interface{}]interface{})
 	if !ok {
-		return nil, errors.New("Labels missing from deployment config")
+		return nil, errs.New("labels missing from deployment config")
 	}
 	spaceLabel, ok := labels["space"].(string)
 	if !ok || len(spaceLabel) == 0 {
-		return nil, errors.New("Space label missing from deployment config")
+		return nil, errs.New("space label missing from deployment config")
 	}
 	if spaceLabel != space {
-		return nil, errors.New("Deployment config " + appName + " is part of space " +
+		return nil, errs.New("deployment config " + appName + " is part of space " +
 			spaceLabel + ", expected space " + space)
 	}
 	// Get UID from deployment config
 	uid, ok := metadata["uid"].(string)
 	if !ok || len(uid) == 0 {
-		return nil, errors.New("Malformed metadata in deployment config")
+		return nil, errs.New("malformed metadata in deployment config")
```

Comment:
I suggest to include the malformed metadata to enrich the error message: `errs.Errorf("malformed metadata in deployment config (not a string): %+v", metadata["uid"])`. This applies to all other errors as well.

Reply:
Done in several cases.

Reply:
@stooke I can't see the change that @kwk suggested